# 01. Importing Libraries

In [1]:
# Import libraries
import pandas as pd
import numpy as np
import os


# 02 Import Data

In [2]:
path = r'C:\Users\tee3a\3D Objects\12-2022 Instacart Basket Analysis'

In [3]:
path

'C:\\Users\\tee3a\\3D Objects\\12-2022 Instacart Basket Analysis'

In [4]:
vars_list = ['order_id', 'user_id', 'order_number', 'order_dow', 'order_hour_of_day', 'days_since_prior_order']

In [5]:
vars_list

['order_id',
 'user_id',
 'order_number',
 'order_dow',
 'order_hour_of_day',
 'days_since_prior_order']

In [6]:
df = pd.read_csv(os.path.join(path, '02 Data', 'Original Data for raw data sets', 'orders.csv'), usecols = vars_list)

In [7]:
df.head(5)

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   2539329        1             1          2                  8   
1   2398795        1             2          3                  7   
2    473747        1             3          3                 12   
3   2254736        1             4          4                  7   
4    431534        1             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

In [8]:
df_prods = pd.read_csv(os.path.join(path, '02 Data', 'Original Data for raw data sets', 'products.csv'), index_col = False)

In [9]:
df_prods.head(20) + df_prods.tail(20)

product_id product_name  aisle_id  department_id  prices
0             NaN          NaN       NaN            NaN     NaN
1             NaN          NaN       NaN            NaN     NaN
2             NaN          NaN       NaN            NaN     NaN
3             NaN          NaN       NaN            NaN     NaN
4             NaN          NaN       NaN            NaN     NaN
5             NaN          NaN       NaN            NaN     NaN
6             NaN          NaN       NaN            NaN     NaN
7             NaN          NaN       NaN            NaN     NaN
8             NaN          NaN       NaN            NaN     NaN
9             NaN          NaN       NaN            NaN     NaN
10            NaN          NaN       NaN            NaN     NaN
11            NaN          NaN       NaN            NaN     NaN
12            NaN          NaN       NaN            NaN     NaN
13            NaN          NaN       NaN            NaN     NaN
14            NaN          NaN       NaN            NaN     NaN
15            NaN          NaN       NaN            NaN     NaN
16            NaN          NaN       NaN            NaN     NaN
17            NaN          NaN       NaN            NaN     NaN
18            NaN          NaN       NaN            NaN     NaN
19            NaN          NaN       NaN            NaN     NaN
49673         NaN          NaN       NaN            NaN     NaN
49674         NaN          NaN       NaN            NaN     NaN
49675         NaN          NaN       NaN            NaN     NaN
49676         NaN          NaN       NaN            NaN     NaN
49677         NaN          NaN       NaN            NaN     NaN
49678         NaN          NaN       NaN            NaN     NaN
49679         NaN          NaN       NaN            NaN     NaN
49680         NaN          NaN       NaN            NaN     NaN
49681         NaN          NaN       NaN            NaN     NaN
49682         NaN          NaN       NaN            NaN     NaN
49683         NaN          NaN       NaN            NaN     NaN
49684         NaN          NaN       NaN            NaN     NaN
49685         NaN          NaN       NaN            NaN     NaN
49686         NaN          NaN       NaN            NaN     NaN
49687         NaN          NaN       NaN            NaN     NaN
49688         NaN          NaN       NaN            NaN     NaN
49689         NaN          NaN       NaN            NaN     NaN
49690         NaN          NaN       NaN            NaN     NaN
49691         NaN          NaN       NaN            NaN     NaN
49692         NaN          NaN       NaN            NaN     NaN

In [10]:
df_prods.columns

Index(['product_id', 'product_name', 'aisle_id', 'department_id', 'prices'], dtype='object')

In [11]:
df_prods.shape

(49693, 5)

# 02 Answer the following questions using the relevant commands. Use markdown mode for your Jupyter cells to provide written answers within your Jupyter notebook after executing the commands

In [12]:
df_prods.describe()

product_id      aisle_id  department_id        prices
count  49693.000000  49693.000000   49693.000000  49693.000000
mean   24844.345139     67.770249      11.728433      9.994136
std    14343.717401     38.316774       5.850282    453.519686
min        1.000000      1.000000       1.000000      1.000000
25%    12423.000000     35.000000       7.000000      4.100000
50%    24845.000000     69.000000      13.000000      7.100000
75%    37265.000000    100.000000      17.000000     11.200000
max    49688.000000    134.000000      21.000000  99999.000000

What’s the maximum value of the “aisle_id” column?  134

In [13]:
df_prods.dtypes

product_id         int64
product_name      object
aisle_id           int64
department_id      int64
prices           float64
dtype: object

What’s the data type of the “department_id” column? int64